packages/application/src/test/application.spec.ts

```diff
@@ -0,0 +1,266 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as nbformat from '@jupyterlab/nbformat';
+import { NotebookModel } from '@jupyterlab/notebook';
+import { JupyterServer, NBTestUtils } from '@jupyterlab/testutils';
+
+import { NotebookParser } from '../parsing';
+import { RequestHandler } from '../requests';
+import { FrontendServices } from '../services';
+
+const server = new JupyterServer();
+const defaultContent: any = {
+  cells: [
+    {
+      cell_type: 'code',
+      execution_count: null,
+      metadata: {},
+      outputs: [],
+      source: ['import os\n', "print(os.environ['HOME'])"]
+    }
+  ],
+  metadata: {
+    kernelspec: {
+      display_name: 'Python 3',
+      language: 'python',
+      name: 'python3'
+    },
+    language_info: {
+      codemirror_mode: {
+        name: 'ipython',
+        version: 3
+      },
+      file_extension: '.py',
+      mimetype: 'text/x-python',
+      name: 'python',
+      nbconvert_exporter: 'python',
+      pygments_lexer: 'ipython3',
+      version: '3.7.6'
+    }
+  },
+  nbformat: 4,
+  nbformat_minor: 4
+};
+const codeSnippetSchema = {
+  $schema: 'http://json-schema.org/draft-07/schema#',
+  title: 'Code Snippet',
+  name: 'code-snippet',
+  display_name: 'Code Snippet',
+  namespace: 'code-snippets',
+  uihints: {
+    title: 'Code Snippets',
+    icon: 'elyra:code-snippet'
+  },
+  properties: {
+    schema_name: {
+      title: 'Schema Name',
+      description: 'The schema associated with this instance',
+      type: 'string',
+      pattern: '^[a-z][a-z0-9-_]*[a-z0-9]$',
+      minLength: 1
+    },
+    display_name: {
+      title: 'Display Name',
+      description: 'The display name of the Code Snippet',
+      type: 'string',
+      minLength: 1
+    },
+    metadata: {
+      description: 'Additional data specific to this Code Snippet',
+      type: 'object',
+      properties: {
+        description: {
+          title: 'Description',
+          description: 'Code snippet description',
+          type: 'string'
+        },
+        language: {
+          title: 'Language',
+          description: 'Code snippet implementation language',
+          type: 'string',
+          uihints: {
+            field_type: 'dropdown',
+            default_choices: ['Python', 'Java', 'R', 'Scala', 'Markdown']
+          },
+          minLength: 1
+        },
+        code: {
+          title: 'Code',
+          description: 'Code snippet code lines',
+          type: 'array',
+          uihints: {
+            field_type: 'code'
+          }
+        }
+      },
+      required: ['language', 'code']
+    }
+  },
+  required: ['schema_name', 'display_name', 'metadata']
+};
+
+const codeSnippetMetadata = {
```

Comment:
the same comment on the naming, although this is more like a maybe versus must change.